File: 2136-earliest-possible-day-of-full-bloom/2136-earliest-possible-day-of-full-bloom.js
/**
 * @param {number[]} plantTime
 * @param {number[]} growTime
 * @return {number}
 */
var earliestFullBloom = function(plantTime, growTime) {
    const plants = plantTime.map((pT,i) => [pT, growTime[i]])
    plants.sort((a, b) => b[1] - a[1])
    let time = 0, res = 0
    for (const [pT, gT] of plants) {
        time += pT
        res = Math.max(res, time + gT)
    }
    return res
};

File: 2136-earliest-possible-day-of-full-bloom/2136-earliest-possible-day-of-full-bloom.py
class Solution:
    def earliestFullBloom(self, plantTime: List[int], growTime: List[int]) -> int:
        
        plants = [[pt, growTime[i]] for i, pt in enumerate(plantTime)]
        plants = sorted(plants, key=lambda x: x[1], reverse=True)
        
        time, res = 0, 0
        
        for pt, gt in plants:
            time += pt
            res = max(res, time + gt)
        
        return res
        